src/middleware/admin.js

```diff
@@ -37,93 +39,69 @@ module.exports = function (middleware) {
-				var templateValues = {
-					config: res.locals.config,
-					configJSON: jsesc(JSON.stringify(res.locals.config), { isScriptContext: true }),
-					relative_path: res.locals.config.relative_path,
-					adminConfigJSON: encodeURIComponent(JSON.stringify(results.configs)),
-					user: userData,
-					userJSON: jsesc(JSON.stringify(userData), { isScriptContext: true }),
-					plugins: results.custom_header.plugins,
-					authentication: results.custom_header.authentication,
-					scripts: results.scripts,
-					'cache-buster': meta.config['cache-buster'] || '',
-					env: !!process.env.NODE_ENV,
-					title: (acpPath || 'Dashboard') + ' | NodeBB Admin Control Panel',
-					bodyClass: data.bodyClass,
-					version: version,
-					latestVersion: results.latestVersion,
-					upgradeAvailable: results.latestVersion && semver.gt(results.latestVersion, version),
-				};
-
-				templateValues.template = { name: res.locals.template };
-				templateValues.template[res.locals.template] = true;
-
-				req.app.render('admin/header', templateValues, next);
-			},
-		], next);
+
+		const results = await utils.promiseParallel({
+			userData: user.getUserFields(req.uid, ['username', 'userslug', 'email', 'picture', 'email:confirmed']),
+			scripts: getAdminScripts(),
+			custom_header: plugins.fireHook('filter:admin.header.build', custom_header),
+			configs: meta.configs.list(),
+			latestVersion: versions.getLatestVersion(),
+		});
+
+		var userData = results.userData;
+		userData.uid = req.uid;
+		userData['email:confirmed'] = userData['email:confirmed'] === 1;
+
+		var acpPath = req.path.slice(1).split('/');
+		acpPath.forEach(function (path, i) {
+			acpPath[i] = path.charAt(0).toUpperCase() + path.slice(1);
+		});
+		acpPath = acpPath.join(' > ');
+
+		var version = nconf.get('version');
+
+		res.locals.config.userLang = res.locals.config.acpLang || res.locals.config.userLang;
+		var templateValues = {
+			config: res.locals.config,
+			configJSON: jsesc(JSON.stringify(res.locals.config), { isScriptContext: true }),
+			relative_path: res.locals.config.relative_path,
+			adminConfigJSON: encodeURIComponent(JSON.stringify(results.configs)),
+			user: userData,
+			userJSON: jsesc(JSON.stringify(userData), { isScriptContext: true }),
+			plugins: results.custom_header.plugins,
+			authentication: results.custom_header.authentication,
+			scripts: results.scripts,
+			'cache-buster': meta.config['cache-buster'] || '',
+			env: !!process.env.NODE_ENV,
+			title: (acpPath || 'Dashboard') + ' | NodeBB Admin Control Panel',
+			bodyClass: data.bodyClass,
+			version: version,
+			latestVersion: results.latestVersion,
+			upgradeAvailable: results.latestVersion && semver.gt(results.latestVersion, version),
+		};
+
+		templateValues.template = { name: res.locals.template };
+		templateValues.template[res.locals.template] = true;
+
+		const render = util.promisify(req.app.render.bind(req.app));
```

Comment:
Should be able to use `req.app.renderAsync`

Reply:
Oh, I didn't know that, thanks 👍